pytools/__init__.py

```diff
@@ -411,9 +411,14 @@ class RecordWithoutPickling:
     """
 
     __slots__: ClassVar[List[str]] = []
-    fields: ClassVar[Set[str]]
+
+    # A dict, not a set, to maintain a deterministic iteration order
+    fields: ClassVar[Dict[str, None]]
 
     def __init__(self, valuedict=None, exclude=None, **kwargs):
+        from warnings import warn
+        warn("pytools.Record is deprecated. Use dataclasses instead.")
```

Comment:
Note that this is `RecordWithoutPickling`.

Reply:
Right, would you prefer something like `pytools.{self.__class__}` or `{pytools.{self.__class__.__bases__[0]}}`  ?